File: src/main/java/com/opdapp/model/DoseFrequency.java
package com.opdapp.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class DoseFrequency {

    @Id
    @GeneratedValue
    private long doseFrequencyId;

    private String timeUnit;
    private int noofDoses;

    public long getDoseFrequencyId() {
        return doseFrequencyId;
    }

    public void setDoseFrequencyId(long doseFrequencyId) {
        this.doseFrequencyId = doseFrequencyId;
    }

    public String getTimeUnit() {
        return timeUnit;
    }

    public void setTimeUnit(String timeUnit) {
        this.timeUnit = timeUnit;
    }

    public int getNoofDoses() {
        return noofDoses;
    }

    public void setNoofDoses(int noofDoses) {
        this.noofDoses = noofDoses;
    }
}

File: frontend/src/app/drugcomponent/dosefrequency.ts
export class Dosefrequency {
  private _doseFrequencyId;
  private _timeUnit;
  private _noofDoses;


  get doseFrequencyId() {
    return this._doseFrequencyId;
  }

  set doseFrequencyId(value) {
    this._doseFrequencyId = value;
  }

  get timeUnit() {
    return this._timeUnit;
  }

  set timeUnit(value) {
    this._timeUnit = value;
  }

  get noofDoses() {
    return this._noofDoses;
  }

  set noofDoses(value) {
    this._noofDoses = value;
  }
}
